standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -1933,6 +1934,16 @@ public List<Partition> dropPartitions(String catName, String dbName, String tblN
                                         PartitionDropOptions options) throws TException {
     RequestPartsSpec rps = new RequestPartsSpec();
     List<DropPartitionsExpr> exprs = new ArrayList<>(partExprs.size());
+    Table table = getTable(catName, dbName, tblName);
+    HiveMetaHook hook = getHook(table);
+    EnvironmentContext context = new EnvironmentContext();
+    if (hook != null) {
+      hook.preDropPartition(table, context, partExprs);
```

Comment:
should we return a boolean (skipDrop) here instead of passing it through the context?

Reply:
This is dependent on hook implementation hence skipping based on a boolean is not the best approach. Thats why passing the skip option through the context.